endpoints/openrtb2/amp_auction.go

```diff
@@ -379,22 +382,21 @@ func (deps *endpointDeps) overrideWithParams(httpRequest *http.Request, req *ope
 		req.Imp[0].TagID = slot
 	}
 
-	//In the AMP endpoint the consent string found in the http.Request query overrides that of the prebid query
-	queryConsentString := httpRequest.FormValue("gdpr_consent")
-	if queryConsentString != "" {
-		jsonMsg := json.RawMessage(`{"consent":"` + queryConsentString + `"}`)
-		// If nil, initialize
-		if req.User == nil {
-			req.User = &openrtb.User{Ext: jsonMsg}
-		} else if req.User.Ext == nil {
-			req.User.Ext = jsonMsg
-		} else { // req.User.Ext != nil, keep whatever is in there and only substitute the consent string
-			var parserErr error
-			req.User.Ext, parserErr = jsonparser.Set(req.User.Ext, []byte(`"`+queryConsentString+`"`), "consent")
-			if parserErr != nil {
-				return parserErr
-			}
-		}
+	gdprConsent, gdprOK := getQueryParam(httpRequest, "gdpr_consent")
+	ccpaValue, ccpaOK := getQueryParam(httpRequest, "us_privacy")
+	privacyPolicies := privacy.Policies{
+		GDPR: gdpr.Policy{
+			Enabled: gdprOK && len(gdprConsent) > 0,
```

Comment:
For compatibility with existing code, an empty GDPR consent value is interpreted as not enabled. For CCPA, I'd like to be less opinionated and pass through exactly the string provided, even if it's empty. Thoughts?

Reply:
I thought GDPR being turned on, but an empty consent string meant that GDPR applied with zero consents given.

Reply:
Oh, yes, we are only sending one GDPR signal in the AMP query, never mind

Reply:
Should look better now with nil having the same meaning as empty.